src/SymfonyCache/CustomTtlListener.php

```diff
@@ -89,12 +99,15 @@ public function cleanResponse(CacheEvent $e)
             } else {
                 $response->headers->addCacheControlDirective('s-maxage', $smaxage);
             }
+
+            $response->headers->remove(static::SMAXAGE_BACKUP);
         }
 
-        if (!$this->keepTtlHeader) {
+        if ($response->headers->has($this->ttlHeader)
```

Comment:
are you sure about checking for existence? removing a header that does not exist is not wrong - and looking at the code i am not sure which is more efficient.

Reply:
I just moved the if down as the early return was confusing for me as I first did read it as `||` instead of `&&` so I removed the early return and moved the has check here. But you are right removing it think would not hurt and I'm not sure what is faster.